nexus/tests/integration_tests/endpoints.rs

```diff
@@ -356,16 +372,8 @@ pub struct VerifyEndpoint {
     /// "/organizations/{organization_name}".
     pub url: &'static str,
 
-    /// Specifies whether an HTTP resource handled by this endpoint is visible
-    /// to unauthenticated or unauthorized users
-    ///
-    /// If it's [`Visibility::Public`] (like "/organizations"), unauthorized
-    /// users can expect to get back a 401 or 403 when they attempt to access
-    /// it.  If it's [`Visibility::Protected`] (like a specific Organization),
-    /// unauthorized users will get a 404.
-    pub visibility: Visibility,
-
-    /// Specifies what HTTP methods are supported for this HTTP resource
+    /// Specifies what HTTP methods are supported for this HTTP resource, along
```

Comment:
This feels a little off to me and I'm trying to figure out why.

Before this change, there was an idea of "visibility" that applied at the HTTP-resource-level. e.g., "/organizations" is visible to everybody, but "/organizations/some-org" is not).  That's not just "can you GET it", but also whether you get a 403 or 404 when you _can't_ get it.  This also enforced some level of API coherence: you could not specify that an unprivileged user is allowed to delete a resource but not see it.  I think this was useful because in practice I expect us to conform to a small number of patterns: "everyone can see and modify this" (like your own ssh keys), "everyone can see but not modify this" (like /images), and "only privileged users can do anything with this" (most other things).  This also enforced some useful invariants like "if you can delete this thing, then you can also see it".

With this change, the idea of "visibility" of an HTTP resource goes away and we're really talking about access for a specific method+URL combination.  This allows allows you to specify some nonsensical configurations now (e.g., you can say that an unauthenticated user can delete a resource but you have to be authenticated to see it).  There's nothing to enforce that the behavior matches one of these patterns.

An alternative might be to have another field here at the endpoint level called "unprivileged_access" or something that specifies what authenticated, unprivileged users are expected to be able to do here.  We could have variants for the patterns that we expect to have: maybe `UnprivilegedAccess::Full`, `UnprivilegedAccess::ReadOnly`, or `UnprivilegedAccess::None` corresponding to the cases I mentioned above.  This would enforce only a handful of possible configurations here, which I think is a good thing because you couldn't (accidentally or otherwise) create the nonsensical configurations I mentioned above.  It'd also be a lot less boilerplate for each endpoint and this change would be much smaller.  What do you think?